### PR TITLE
`azurerm_windows_web_app`,`azurerm_windows_web_app_slot`, `azurerm_windows_function_app`, `azurerm_windows_function_app_slot`, `azurerm_linux_web_app`,`azurerm_linux_web_app_slot`, `azurerm_linux_function_app`, `azurerm_linux_function_app_slot` - add `hosting_environment_id` property

### PR DESCRIPTION
**Pull request by xiaxyi**

### CODE REVIEW

File: internal/services/appservice/windows_function_app_resource.go [hunk 1]
```diff
@@ -639,6 +645,10 @@ func (r WindowsFunctionAppResource) Read() sdk.ResourceFunc {
 				DefaultHostname:             utils.NormalizeNilableString(props.DefaultHostName),
 			}
 
+			if hostingEnv := props.HostingEnvironmentProfile; hostingEnv != nil {
+				state.HostingEnvId = utils.NormalizeNilableString(hostingEnv.ID)
```
Comment (jackofallops) [inline internal/services/appservice/windows_function_app_resource.go:649]: Can we change this to use the `pointer` package as it's already imported? We'll be refactoring all references of this soon, but we shouldn't continue use of  the `utils` package in new changes.

File: internal/services/appservice/windows_web_app_data_source.go [hunk 2]
```diff
@@ -314,6 +320,9 @@ func (d WindowsWebAppDataSource) Read() sdk.ResourceFunc {
 				if subnetId := utils.NormalizeNilableString(props.VirtualNetworkSubnetID); subnetId != "" {
 					webApp.VirtualNetworkSubnetID = subnetId
 				}
+				if hostingEnv := props.HostingEnvironmentProfile; hostingEnv != nil {
+					webApp.HostingEnvId = utils.NormalizeNilableString(hostingEnv.ID)
```
Comment (jackofallops) [inline internal/services/appservice/windows_web_app_data_source.go:324]: As above, can we change to `pointer` package here?

File: website/docs/d/linux_function_app.html.markdown [hunk 3]
```diff
@@ -63,6 +63,8 @@ In addition to the Arguments listed above - the following Attributes are exporte
 
 * `default_hostname` - The default hostname of the Linux Function App.
 
+* `hosting_environment_id` - The hosting environment id used by Function App.
```
Comment (jackofallops) [inline website/docs/d/linux_function_app.html.markdown:66]: The resource name in the provider is `app_servce_environment` we should make this clearer in the documentation references? Perhaps:
 ```suggestion
* `hosting_environment_id` - The ID of the App Service Environment used by this Linux Function App.
```
Can you also do similar for the other property references below?

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @xiaxyi - Thanks for this PR.  I've left a few comments inline below if you can take a look and address I think this will be good to merge.

Thanks!

**Comment by xiaxyi** [comment]
Thanks @jackofallops , code is updated.

**Review by jackofallops — APPROVED** [review]
Thanks for those changes @xiaxyi - this LGTM now 👍

**Pull request merged**